qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,346 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class for calibration-type experiments."""
+
+from abc import ABC
+from typing import Any, Dict, Iterable, Optional, Tuple
+
+from qiskit.providers.options import Options
+from qiskit.providers.backend import Backend
+from qiskit.circuit import Parameter
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.framework.base_experiment import BaseExperiment
+from qiskit_experiments.framework.experiment_data import ExperimentData
+from qiskit_experiments.exceptions import CalibrationError
+
+
+class BaseCalibrationExperiment(BaseExperiment, ABC):
```

Comment:
```suggestion
class CalibrationMixin(ABC):
```
check abstract mix-in?